### ISSUE TITLE
Issues with ncc and GraphQL project?

### ISSUE THREAD

**Issue by Siyfion** [zeit/ncc#284]
Not sure if I'm missing something entirely but, if I run my application normally, it all works fine. If I try bundling it up with ncc, I get the following errors:

![screenshot 2019-02-15 at 10 52 10](https://user-images.githubusercontent.com/338488/52852179-c754ba80-310f-11e9-9824-76e309dfddf1.png)

**Comment by guybedford** [comment]
Thanks for posting an issue. Can you share the code that is on line 42980 of dist/index.js that is giving the error? It would also help to know what the original require looked like for that code that isn't working correctly.

**Comment by Siyfion** [comment]
Annoyingly, it's just these lines:

```js
function webpackEmptyContext(req) {
	var e = new Error("Cannot find module '" + req + "'");
	e.code = 'MODULE_NOT_FOUND';
	throw e;
}
```

**Comment by Siyfion** [comment]
Seems like it's these lines (the last one throws the error):
```
var resolveCwd = __webpack_require__(3370);
var graphqlPackagePath = resolveCwd.silent('graphql');
var _a = __webpack_require__(1604)(graphqlPackagePath || 'graphql'), GraphQLObjectType = _a.GraphQLObjectType, GraphQLScalarType = _a.GraphQLScalarType, parse = _a.parse, validate = _a.validate, Kind = _a.Kind;

```

**Comment by Siyfion** [comment]
@guybedford I'm happy to try debugging this further if you can give me any tips / ideas as to where to start?

**Comment by guybedford** [comment]
@Siyfion thanks so much for the updates here, that helps a lot to track this down.

So the issue is exactly that we have code like:

```js
var resolveCwd = require('resolve-cwd');
var graphqlPackagePath = resolveCwd.silent('graphql');
var _a = require(graphqlPackagePath || 'graphql'), GraphQLObjectType = _a.GraphQLObjectType, GraphQLScalarType = _a.GraphQLScalarType, parse = _a.parse, validate = _a.validate, Kind = _a.Kind;
```

where the builder doesn't know the value of `graphqlPackagePath` and therefore we build it in as an unknown module.

One possible fix would be to rewrite requires of the form `require(unknown || 'known-string')` to `require('known-string')`, which seems like it would actually fix this issue for you.

I'll see if I can push up a fix shortly, and link it here for you to test further in due course.

**Comment by Siyfion** [comment]
Thanks @guybedford! If you want me to test out a WIP version, I'm more than happy to be a lab rat! 👍

**Comment by guybedford** [comment]
I've created https://github.com/zeit/ncc/pull/294 for this. Would value a full replication case if you can share to be sure this captures it.

**Comment by Siyfion** [comment]
@guybedford yup, that fixes our issue! 🎉 I can't share this reproduction repo, as it's a client's code, but I'll try to see if I can find something open source that I can link that'll demonstrate too.

**Comment by Siyfion** [comment]
@guybedford found one for you:

https://github.com/prisma/graphql-yoga/tree/master/examples/prisma-js

**Comment by kcjpop** [comment]
Thank you all for your effort. I'm having the exact same issue and cannot deploy new code to our prod. Is there any way to get #294 merged soon?

**Comment by styfle** [comment]
Fixed in #294
